site/en/docs/devtools/javascript/reference/index.md

```diff
@@ -202,6 +202,66 @@ represents which function DevTools is currently highlighting.
+
+A stack frame is a function call together with its argument data. However, frame restart doesn't reset the arguments. In other words, the restart doesn't restore the initial state at function call. Instead, it simply moves the execution pointer to the start of the function and allows you to re-run the statements inside the function's body.
+
+{% Aside 'caution' %}
+During live editing, don't change the order of the call stack frames. This may cause unexpected errors.
+{% endAside %}
+
+To restart a frame:
+
+1. [Pause function execution at a breakpoint](#breakpoints). The **Call Stack** pane records the order of function calls.
+1. In the **Call Stack** pane, right-click a function and select **Restart frame** from the drop-down menu.
+
+   {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/h54JUoqTr2AvSNesZQI0.png", alt="Selecting Restart frame from the drop-down menu.", width="800", height="497" %}
+
+   {% Aside %}
+   **Note**: You can restart any function frame in the **Call Stack**, except WebAssembly, async, and generator functions.
+   {% endAside %}
+
+To understand how **Restart frame** works, consider the following code:
+
+```js
+function foo(value) {
+    console.log(value);
+    bar(value);
+}
+ 
+function bar(value) {
+    value++;
+    console.log(value);
+    debugger;
+}
+
+foo(0);
+```
+
+The `foo()` function takes `0` as an argument, logs it, and calls the `bar()` function. The `bar()` function, in turn, increments the argument.
+
+Try restarting the frames of both functions in the following way:
+
+1. Copy the code to a [new snippet](/docs/devtools/javascript/snippets/#createsources) and [run it](/docs/devtools/javascript/snippets/#runsources). The execution stops at the `debugger` [line-of-code breakpoint](/docs/devtools/javascript/breakpoints/#debugger).
+1. Notice that the debugger shows you the current value next to function declaration: `value = 1`.
+   {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/i3Offlw9RToaew8APV4C.png", alt="The current value next to function declaration.", width="800", height="497" %}
+1. Restart the `bar()` frame.
+   {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/sMDXhnV3Ft02evS0PBQR.png", alt="Restarting the bar() frame.", width="800", height="497" %}
+1. Step through the value increment statement by pressing `F9`.
+   {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/jno28U2OaVMnc2s6xtRZ.png", alt="Incrementing current value.", width="800", height="497" %}
+   Notice that the current value increases: `value = 2`.
+1. Try restarting the `bar()` frame several more times. The value continues to increase.
+   {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/LGHUF27jZmP341zxOLZB.png", alt="Restarting the bar() frame again.", width="800", height="497" %}
+   The debugger doesn't "remember" the initial argument value. The current value persists in memory across restarts of the same function.
```

Comment:
Make this a "gotcha - why the value doesn't reset to 0"? This reflect your explanation at the beginning "In other words, the restart doesn't restore the initial state at function call. Instead, it simply moves the execution pointer to the start of the function and allows you to re-run the statements inside the function's body".